packages/core/content-type-builder/admin/src/utils/parseDateValue.ts

```diff
@@ -0,0 +1,18 @@
+import { Schema } from '@strapi/types';
+
+export const parseDateValue = <TAttribute extends Schema.Attribute.AnyAttribute>(
+  value: Schema.Attribute.Value<TAttribute>
+): Date | undefined => {
+  if (value instanceof Date) {
+    return isValidDate(value) ? value : undefined;
+  }
+
+  if (typeof value === 'string' || typeof value === 'number') {
+    const date = new Date(value);
+    return isValidDate(date) ? date : undefined;
+  }
+
+  return undefined;
```

Comment:
no need for the `return undefined` this will be the default behavior of the method